### PR TITLE
Support V5P for type of accelerator_config in google_tpu_v2_vm

### PR DESCRIPTION
**Pull request by marblejenka**
This is PR for https://github.com/hashicorp/terraform-provider-google/issues/17734 .

```release-note:enhancement
tpu: added `type` of `accelerator_config` to `google_tpu_v2_vm` resource
```

### CODE REVIEW

**Comment by BBBmau** [comment]
@marblejenka we ran into some issues with CI, could you rebase to cover these issues? We can then run the tests before being able to approve.

**Comment by marblejenka** [comment]
I found that the tests in the local environment were not running properly, so I will make some corrections. Please wait for a while.

File: mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+resource "google_tpu_v2_vm" "<%= ctx[:primary_resource_id] %>" {
+  provider = google-beta
+
+  name = "<%= ctx[:vars]["vm_name"] %>"
+  zone = "us-east5-a"
```
Comment (marblejenka) [inline mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb:5]: At this time, it appears that it is only available the zone for me due to quota restriction.

File: mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  runtime_version = "tpu-vm-tf-2.16.1-pjrt"
+
+  accelerator_config {
+    type     = "V5P"
+    topology = "2x2x1"
+  }
+
+  network_config {
+    enable_external_ips = true
```
Comment (marblejenka) [inline mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb:15]: This configuration is essentially unnecessary, but I added it to work around the constraints of the VPC network when running tests.

File: mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  scheduling_config {
+    preemptible = true
```
Comment (marblejenka) [inline mmv1/templates/terraform/examples/tpu_v2_vm_v5p.tf.erb:19]: This configuration is essentially unnecessary, but I added it to work around the quota restriction when running tests.

**Comment by marblejenka** [comment]
@BBBmau The test is failing with `Error 403: Location us-east5-a is not found or access is unauthorized.`. From what I've tried, `us-east5-a` seems to be the only region where we can use TPU v5p with QIR. Can you change settings for the testing Google Cloud project?

The requirement is to increase  TPUV5PPreemptiblePerProjectPerZoneForTPUAPI quota to 8 or higher (default value is 0) mabye in us-east5-a.

**Comment by bollenberger** [comment]
@BBBmau it looks like in general we've not added accelerator quota for each generation to this test project. The PR overall looks good and we can actually do without the additional test example for the time being, as we don't have access to long term Terraform testing quota.

I'd recommend removing the new example and approving the PR otherwise.

**Comment by marblejenka** [comment]
@BBBmau @bollenberger Finally, I deleted all the codes except "V5P". [This document](https://cloud.google.com/tpu/docs/reference/rest/v2/AcceleratorConfig) suggests that this PR works well.

**Review by BBBmau — APPROVED** [review]

**Comment by marblejenka** [comment]
@BBBmau Can you merge this one? I don't have the permission to merge.

**Pull request merged**